examples/mini_repl.rs

```diff
@@ -0,0 +1,54 @@
+///! This example show cases a very simple REPL.
+///! While a much better REPL can be found in ../src/shell,
+///! This much smaller REPL is still a useful example because it showcases inserting
+///! values and functions into the Python runtime's scope, and showcases use
+///! of the compilation mode "Single".
+///! Note that in particular this REPL does a horrible job of showing users their errors
+///! (instead it simply crashes).
+use rustpython_compiler::compile;
+use rustpython_vm::{pyobject::PyResult, PySettings, VirtualMachine};
+// this needs to be in scope in order to insert things into scope.globals
+use rustpython_vm::pyobject::ItemProtocol;
+
+fn main() -> PyResult<()> {
+    let mut on = true;
+
+    let mut input = String::with_capacity(50);
+    let stdin = std::io::stdin();
+
+    let vm = VirtualMachine::new(PySettings::default());
+    let scope = vm.new_scope_with_builtins();
+
+    // typing `quit()` is too long, let's make `on(False)` work instead.
+    scope
+        .globals
+        .set_item(
+            "on",
+            vm.context().new_rustfunc({
+                let on: *mut bool = &mut on;
+                move |b: bool, _: &VirtualMachine| unsafe { *on = b }
+            }),
```

Comment:
We actually renamed `new_rustfunc` to `new_function` recently; with the above change this can just be `vm.context().new_function(|b: bool| on.set(b))`.

Reply:
I still needed a separate scope in which to clone the `Rc`, but thanks for the tip about `new_function`!